### PR TITLE
PIM-10001: Merge 4.0 into 5.0

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/job_constraints.yml [hunk 1]
```diff
@@ -22,10 +22,9 @@ services:
         tags:
             - { name: akeneo_batch.job.job_parameters.constraint_collection_provider }
 
-    akeneo_pim_enrichment.job.job_parameters.constraint_collection_provider.clean_removed_attribute:
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/Resources/config/job_constraints.yml:25]: The `akeneo_pim_enrichment.job.job_parameters.constraint_collection_provider.clean_removed_attribute` should be kept IMO

File: src/Akeneo/Platform/Bundle/InstallerBundle/Resources/fixtures/minimal/jobs.yml [hunk 2]
```diff
@@ -135,8 +135,8 @@ jobs:
         alias:     remove_non_existing_product_values
         label:     Remove non existing product values
         type:      remove_non_existing_product_values
-    clean_removed_attribute_job:
```
Comment (nmarniesse) [inline src/Akeneo/Platform/Bundle/InstallerBundle/Resources/fixtures/minimal/jobs.yml:138]: Are you sure this job needs to be removed?

File: tests/back/Integration/catalog/technical/jobs.yml [hunk 3]
```diff
@@ -188,8 +188,8 @@ jobs:
         alias:     remove_non_existing_product_values
         label:     Remove non existing product values
         type:      remove_non_existing_product_values
-    clean_removed_attribute_job:
```
Comment (nmarniesse) [inline tests/back/Integration/catalog/technical/jobs.yml:191]: Are you sure this job needs to be removed?

File: tests/legacy/features/Context/catalog/footwear/jobs.yml [hunk 4]
```diff
@@ -578,8 +578,8 @@ jobs:
         alias:     remove_non_existing_product_values
         label:     Remove non existing product values
         type:      remove_non_existing_product_values
-    clean_removed_attribute_job:
```
Comment (nmarniesse) [inline tests/legacy/features/Context/catalog/footwear/jobs.yml:581]: same here

**Comment by anaisbaunelemaire** [comment]
Thanks @nmarniesse! I made mistakes, indeed...

**Review by RiffVentura — APPROVED** [review]

**Review by rfauglas — APPROVED** [review]

**Comment by RiffVentura** [comment]
warm

**Pull request merged**